src/dividat-driver/server/main.go

```diff
@@ -107,3 +107,25 @@ func Start(logger *logrus.Logger) context.CancelFunc {
 
 	return cancel
 }
+
+// Middleware for CORS headers, to be applied to any route that should be accessible from browser apps.
+func corsHeaders(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if len(r.Header["Origin"]) == 1 && isPermissibleOrigin(r.Header["Origin"][0]) {
+			w.Header().Set("Access-Control-Allow-Origin", r.Header["Origin"][0])
+			w.Header().Set("Access-Control-Allow-Private-Network", "true")
```

Comment:
We include this header in the server response if the `Origin` sent by the client is among the permissible/expected origins to contact the driver's endpoints. The client need not sent the `Access-Control-Request-Private-Network` header, but if it is missing in the response it may choose not to complete the request (it would instead fail after the preflight request).